package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -0,0 +1,821 @@
+            self.logger_debug("Water donors: {0}".format(len(self._water_donors)))
+            self.logger_debug("Water donor hydrogens: {0}".format(len(self._water_donors_h)))
+            self._water_acceptors = self._water.select_atoms(
+                'name {0}'.format(' '.join(self.acceptors)))
+            self.logger_debug("Water acceptors: {0}".format(len(self._water_acceptors)))
+
+    def run(self, **kwargs):
+        """Analyze trajectory and produce timeseries.
+
+        Stores the water bridge data per frame as
+        :attr:`WaterBridgeAnalysis.timeseries` (see there for output
+        format).
+
+        Parameters
+        ----------
+        verbose : bool (optional)
+             toggle progress meter output :class:`~MDAnalysis.lib.log.ProgressMeter`
+             [``True``]
+        debug : bool (optional)
+             enable detailed logging of debugging information; this can create
+             *very big* log files so it is disabled (``False``) by default; setting
+             `debug` toggles the debug status for :class:`WaterBridgeAnalysis`,
+             namely the value of :attr:`WaterBridgeAnalysis.debug`.
+
+        See Also
+        --------
+        :meth:`WaterBridgeAnalysis.generate_table` :
+               processing the data into a different format.
+        """
+        logger.info("WBridge analysis: starting")
+        logger.debug("WBridge analysis: donors    %r", self.donors)
+        logger.debug("WBridge analysis: acceptors %r", self.acceptors)
+        logger.debug("WBridge analysis: water bridge %r", self.water_selection)
+
+        remove_duplicates = kwargs.pop('remove_duplicates', True)  # False: old behaviour
+        if not remove_duplicates:
+            logger.warning("Hidden feature remove_duplicates=False activated: you will probably get duplicate H-bonds.")
+
+        debug = kwargs.pop('debug', None)
+        if debug is not None and debug != self.debug:
+            self.debug = debug
+            logger.debug("Toggling debug to %r", self.debug)
+        if not self.debug:
+            logger.debug("WBridge analysis: For full step-by-step debugging output use debug=True")
+
+        self._timeseries = []
+        self.timesteps = []
+
+        logger.info("checking trajectory...")  # n_frames can take a while!
+        try:
+            frames = np.arange(self.u.trajectory.n_frames)[self.traj_slice]
+        except:
+            logger.error("Problem reading trajectory or trajectory slice incompatible.")
+            logger.exception()
+            raise
+        verbose = _set_verbose(verbose=kwargs.get('verbose', None),
+                               quiet=kwargs.get('quiet', None),
+                               default=True)
+        pm = ProgressMeter(len(frames),
+                           format="WBridge frame {current_step:5d}: {step:5d}/{numsteps} [{percentage:5.1f}%]\r",
+                           verbose=verbose)
+
+        try:
```

Comment:
this should be `self.u.trajectory.ts.time` and doesn't need wrapping in a try/except block, it's in the base API

Reply:
@richardjgowers 
Thank you for your comment. I guess this try and catch assumes that there are some cases where calling self.u.trajectory.ts.time will invoke NotImplementedError. I guess this issue is probably resolved now.